src/devices/Seesaw/SeesawBase.cs

```diff
@@ -0,0 +1,176 @@
+        /// </summary>
+        /// <param name="i2cDevice">The I2cDevice to be used to communicate with the SeeSaw module. Note that the I2cDevice
+        /// will be disposed when the along with the SeeSaw device</param>
+        public Seesaw(I2cDevice i2cDevice)
+        {
+            Initialize(i2cDevice);
+        }
+
+        /// <summary>
+        /// Version of the SeeSaw module. 
+        /// </summary>
+        public UInt32 Version { get; private set; }
+
+        /// <summary>
+        /// Reads the temperature of the SeeSaw device. 
+        /// </summary>
+        /// <returns>A float that represents the temperature in degrees celcius.</returns>
+        public float GetTemperature()
+        {
+            byte[] buffer = Read(Seesaw_Module.Status, Seesaw_Function.Status_Temp, 4, 1000);
+
+            return ((1.0F / (1UL << 16)) * (((UInt32)buffer[0] << 24) | ((UInt32)buffer[1] << 16) | ((UInt32)buffer[2] << 8) | (UInt32)buffer[3]));
+        }
+
+        /// <summary>
+        /// Tests to see if a module has been compiled into the SeeSaw firmware. 
+        /// </summary>
+        /// <param name="moduleAddress">>An Seesaw_Module enum that represents the mdule to test for.</param>
+        /// <returns>Returns true if the functionality associated with the module is available.</returns>
+        public bool HasModule(Seesaw_Module moduleAddress)
+        {
+            return ((_options & (1 << (byte)moduleAddress)) != 0);
+        }
+
+        /// <summary>
+        /// Performs a soft reset of the SeeSaw module. 
+        /// </summary>
+        public void SWReset()
+        {
+            WriteByte(Seesaw_Module.Status, Seesaw_Function.Status_Swrst, 0xFF);
+        }
+
+        /// <summary>
+        /// Initializes the Seesaw device. 
+        /// <param name="i2cDevice">The I2cDevice to initialize the Seesaw device with.</param>
+        /// </summary>
+        protected void Initialize(I2cDevice i2cDevice)
+        {
+            I2cDevice = i2cDevice;
+
+            SWReset();
+
+            Task.Delay(10).Wait();
+
+            if(ReadByte(Seesaw_Module.Status, Seesaw_Function.Status_Hw_Id) != Sessaw_Hardware_Id)
+            {
+                throw new NotSupportedException($"The hardware on I2C Bus {I2cDevice.ConnectionSettings.BusId}, Address 0x{I2cDevice.ConnectionSettings.DeviceAddress:X2} does not appear to be an Adafruit SeeSaw module");
+            }
+
+            _options = GetOptions();
+            Version = GetVersion();
+        }
+
+        /// <summary>
+        /// Get the firmware version of the Seesaw board. 
+        /// </summary>
+        /// <returns>Returns the seesaw version.</returns>
+        protected UInt32 GetVersion()
+        {
+            byte[] buffer = Read(Seesaw_Module.Status, Seesaw_Function.Status_Version, 4);
+
+            return (((UInt32)buffer[0] << 24) | ((UInt32)buffer[1] << 16) | ((UInt32)buffer[2] << 8) | (UInt32)buffer[3]);
```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.buffers.binary.binaryprimitives?view=netstandard-2.1

Reply:
Done in the latest PR update.